<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>722</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_4">
   <item>
    <widget class="QGroupBox" name="groupBox_4">
     <property name="title">
      <string>文法输入</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <item>
         <widget class="QPushButton" name="open">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>打开</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="save">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>保存</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QTextEdit" name="grammarInput">
        <property name="styleSheet">
         <string notr="true"/>
        </property>
        <property name="placeholderText">
         <string>请输入文法</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_5">
     <item>
      <widget class="QGroupBox" name="groupBox">
       <property name="title">
        <string>功能列表</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QPushButton" name="simplifyGrammar">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>化简文法</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="eliminateLeftRecursion">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>消除左递归</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="leftFactoring">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>提取左公因子</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QPushButton" name="firstSet">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>显示first集合</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="followSet">
            <property name="styleSheet">
             <string notr="true"/>
            </property>
            <property name="text">
             <string>显示follow集合</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_2">
       <property name="title">
        <string>句子输入</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_3">
        <item>
         <widget class="QLineEdit" name="sentenceInput">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="placeholderText">
           <string>请输入待分析的句子</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="leftmostDerive">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
          <property name="text">
           <string>对句子进行最左推导</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="groupBox_3">
       <property name="title">
        <string>最左推导过程</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QListWidget" name="listWidget">
          <property name="styleSheet">
           <string notr="true"/>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
